mambaSharedFramework/HLS Models/Playlist Structure/HLSPlaylistStructure.swift

```diff
@@ -244,9 +244,16 @@ final class HLSPlaylistStructure: HLSPlaylistStructureInterface {
         do {
             let result = try HLSPlaylistStructureConstructor.generateMediaGroups(fromTags: _tags)
             
-            let mediaSpans = try HLSPlaylistStructureConstructor.generateMediaSpans(fromTags: _tags,
-                                                                                    header: result.header,
-                                                                                    mediaSegmentGroups: result.mediaSegmentGroups)
+            let mediaSpans: [TagSpan]
+            do {
+                mediaSpans = try HLSPlaylistStructureConstructor.generateMediaSpans(
```

Comment:
sort of a matter of style but if we're not going to do anything with the thrown error, then I think the `try?` operator would be more appropriate and then you could nil coalesce that to `[]`. Fully realize that with a beefy initializer this may look weird.

Reply:
Oh yeah, that's much nicer!